test/e2e/parallel/spacerequest_test.go

```diff
@@ -0,0 +1,172 @@
+package parallel
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestCreateSpaceRequest(t *testing.T) {
+	// given
+	t.Parallel()
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+	memberCluster, found, err := hostAwait.GetToolchainCluster(t, cluster.Member, memberAwait.Namespace, nil)
+	require.NoError(t, err)
+	require.True(t, found)
+
+	t.Run("create space request", func(t *testing.T) {
+		// when
+		spaceRequest, subSpace := CreateSpaceRequest(t, awaitilities, memberAwait.ClusterName,
+			WithSpecTierName("appstudio"),
+			WithSpecTargetClusterRoles([]string{cluster.RoleLabel(cluster.Tenant)}))
+		// then
+		subSpace, _ = VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name, UntilSpaceHasAnyTargetClusterSet())
+		spaceRequest, err := memberAwait.WaitForSpaceRequest(t, types.NamespacedName{Namespace: spaceRequest.GetNamespace(), Name: spaceRequest.GetName()},
+			UntilSpaceRequestHasConditions(Provisioned()),
+			UntilSpaceRequestHasStatusTargetClusterURL(memberCluster.Spec.APIEndpoint))
+		require.NoError(t, err)
+
+		t.Run("subSpace is recreated if deleted ", func(t *testing.T) {
+			// now, delete the subSpace, along with its associated namespace,
+			// but a new Space will be provisioned by the SpaceRequest.
+
+			// when
+			err := memberAwait.Client.Delete(context.TODO(), subSpace)
+
+			// then
+			require.NoError(t, err)
+			err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(t, subSpace.Name)
+			require.NoError(t, err)
+			err = memberAwait.WaitUntilNSTemplateSetDeleted(t, subSpace.Name)
+			require.NoError(t, err)
```

Comment:
I'm not sure this can happen, unless I'm missing something, the new space that will be created will have a different name, thus checking `err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(t, subSpace.Name)` with the `subSpace.Name` should not find it unless it's still there.

But maybe I'm not understanding it correctly ...

Reply:
you are totally correct :+1: 
I totally forgot that the name of the Space would be different - my bad, sorry for confusion